habitat_sim/agent/agent.py

```diff
@@ -142,9 +142,10 @@ def reconfigure(
         if reconfigure_sensors:
             self._sensors.clear()
             for spec in self.agent_config.sensor_specifications:
-                self._sensors.add(
-                    hsim.CameraSensor(self.scene_node.create_child(), spec)
-                )
+                self.add_sensor(spec)
+
+    def add_sensor(self, spec: hsim.SensorSpec) -> None:
```

Comment:
Should the agent config be updated here for dynamically adding the Sensor?

Reply:
Yeah. Also makes sense to add an assert that the sensor uuid isn't already in the config.